static/app/components/searchSyntax/utils.tsx

```diff
@@ -333,6 +351,12 @@ export function stringifyToken(token: TokenResult<Token>): string {
       return `${token.prefix}[${token.key.value},number]`;
     case Token.KEY_EXPLICIT_STRING_TAG:
       return `${token.prefix}[${token.key.value},string]`;
+    case Token.KEY_EXPLICIT_FLAG:
+      return `flags[${token.key.value}]`;
+    case Token.KEY_EXPLICIT_NUMBER_FLAG:
+      return `flags[${token.key.value},number]`;
+    case Token.KEY_EXPLICIT_STRING_FLAG:
+      return `flags[${token.key.value},string]`;
```

Comment:
This syntax is 😿

How does this search grammar interact with our autocomplete (if at all)? I don't expect anyone to know this (though I suppose it's part of our tag syntax).

It's probably not common for people to use numbers as flag names, but since this syntax already exists, my strong feelings about removing it are less so.